chalice/deploy/newdeployer.py

```diff
@@ -0,0 +1,472 @@
+                filename = os.path.join(config.project_dir,
+                                        '.chalice',
+                                        config.iam_policy_file)
+            else:
+                filename = os.path.join(config.project_dir,
+                                        '.chalice',
+                                        'policy-%s.json' % stage_name)
+            policy = models.FileBasedIAMPolicy(filename=filename)
+        else:
+            resource_name = 'default-role'
+            role_name = '%s-%s' % (config.app_name, stage_name)
+            policy = models.AutoGenIAMPolicy(
+                document=models.Placeholder.BUILD_STAGE)
+        return models.ManagedIAMRole(
+            resource_name=resource_name,
+            role_arn=models.Placeholder.DEPLOY_STAGE,
+            role_name=role_name,
+            trust_policy=LAMBDA_TRUST_POLICY,
+            policy=policy,
+        )
+
+    def _build_lambda_function(self,
+                               config,      # type: Config
+                               function,    # type: app.LambdaFunction
+                               deployment,  # type: models.DeploymentPackage
+                               role,        # type: models.IAMRole
+                               ):
+        # type: (...) -> models.LambdaFunction
+        function_name = '%s-%s-%s' % (
+            config.app_name, config.chalice_stage, function.name)
+        return models.LambdaFunction(
+            resource_name=function.name,
+            function_name=function_name,
+            environment_variables=config.environment_variables,
+            runtime=config.lambda_python_version,
+            handler=function.handler_string,
+            tags=config.tags,
+            timeout=config.lambda_timeout,
+            memory_size=config.lambda_memory_size,
+            deployment_package=deployment,
+            role=role,
+        )
+
+
+class DependencyBuilder(object):
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def build_dependencies(self, graph):
+        # type: (models.Model) -> List[models.Model]
+        seen = set()  # type: Set[int]
+        ordered = []  # type: List[models.Model]
+        for resource in graph.dependencies():
+            self._traverse(resource, ordered, seen)
+        return ordered
+
+    def _traverse(self, resource, ordered, seen):
+        # type: (models.Model, List[models.Model], Set[int]) -> None
+        for dep in resource.dependencies():
+            if id(dep) not in seen:
+                seen.add(id(dep))
+                self._traverse(dep, ordered, seen)
+        if resource not in ordered:
+            ordered.append(resource)
+
+
+class BaseDeployStep(object):
+
+    def handle(self, config, resource):
+        # type: (Config, models.Model) -> None
+        name = 'handle_%s' % resource.__class__.__name__.lower()
+        handler = getattr(self, name, None)
+        if handler is not None:
```

Comment:
If handler is `None` do we want it to silently do nothing? Is there any case where it should be `None`?

Reply:
Being `None` is ok, it just means that there's no action associated with that resource for whatever step we're processing.  For example, not every resource is going to have default values injected.